File: au-java/src/main/java/com/fran6k/study/springbootstudy/bean/WxBean/InterViewEnum.java
package com.fran6k.study.springbootstudy.bean.WxBean;

public enum InterViewEnum {
    Stage_One_WaitB(1,"等待社长审核"),
    Stage_Two_Success(2,"审核通过，等待学生确认"),
    Stage_Three_Success(3,"加入成功"),
    Stage_Two_Failed(4,"审核未通过"),
    Stage_Three_Failed(5,"放弃加入");

    private final int step;
    private final String message;
    private InterViewEnum(int step, String message){
        this.step = step;
        this.message = message;
    }
    public static String getMessage(int step){
        for(InterViewEnum interViewEnum : InterViewEnum.values()){
            if(step == interViewEnum.getStep())return interViewEnum.getMessage();
        }
        return null;
    }
    public static InterViewEnum getByStep(int step){
        for(InterViewEnum interViewEnum : InterViewEnum.values()){
            if(step == interViewEnum.getStep())return interViewEnum;
        }
        return null;
    }
    public int getStep() {
        return step;
    }
    public String getMessage() {
        return message;
    }
}

File: au-association-go/global/InterViewEnum.go
package global

type InterViewEnum struct {
	step    int
	message string
}

var StageOneWaitB = InterViewEnum{step: 1, message: "等待社长审核"}
var StageTwoSuccess = InterViewEnum{step: 2, message: "审核通过，等待学生确认"}
var StageThreeSuccess = InterViewEnum{step: 3, message: "加入成功"}
var StageTwoFailed = InterViewEnum{step: 4, message: "审核未通过"}
var StageThreeFailed = InterViewEnum{step: 5, message: "放弃加入"}

func (interviewEnum *InterViewEnum) GetStep() int {
	return interviewEnum.step
}

func (interviewEnum InterViewEnum) GetMessageByStage(num int) string {
	return InterViewEnum{}.GetByStep(num).message
}

func (interviewEnum InterViewEnum) GetByStep(step int) InterViewEnum {
	switch step {
	case 1:
		return StageOneWaitB
	case 2:
		return StageTwoSuccess
	case 3:
		return StageThreeSuccess
	case 4:
		return StageTwoFailed
	case 5:
		return StageThreeFailed
	}
	return InterViewEnum{}
}
